MCP_7582

amwarnock opened the pull request:
# Description 
The purpose of this ticket is to address comments in [MCP-7582](https://amida.atlassian.net/browse/MCP-7582).  I believe this ticket also sufficiently addresses the problems outlined in [MCP-7583](https://amida.atlassian.net/browse/MCP-7583) as well.

## What was the problem? 
### MCP-7582
SNOMED-CT codes are notoriously robust, encompassing a large variety of diagnoses and procedures.  To assist with data organization, it has been requested that the UMLS script include Semantic Types and Semantic Group data to better differentiate between SNOMED-CT codes that are diagnoses vs procedures and further support SNOMED-CT code-cleaning efforts. 
### MCP-7583
The column organization on output is not optimized for copy/pasting into our current template. 

## How does this fix it? 
MCP-7582
- Add in filtering by Semantic Type and Semantic Group 

MCP-7583
- Semantic Type/Group ordering IDs procedures vs diagnoses
- Updated the output column organization to reflect our template

## Jira Tickets
- [MCP-7582](https://amida.atlassian.net/browse/MCP-7582)
- [MCP-7583](https://amida.atlassian.net/browse/MCP-7583)

## How to Test This PR
1. Choose a condition keyword set of your choice
2. Run the script using the condition keyword set

### Msc. Notes
- I tested this with Lymphoma and a random subset of Animal codes.
- Laura and Hannah's GitHubs are not appearing on my list of able-to-request review.  Making a note here with a request for: 
- 1. If people could also tag them and/or for me to reach out to them on Monday. 

[MCP-7582]: https://amida.atlassian.net/browse/MCP-7582?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MCP-7583]: https://amida.atlassian.net/browse/MCP-7583?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MCP-7582]: https://amida.atlassian.net/browse/MCP-7582?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

jlpeterson2 reviewed (changes requested):
As noted in the jira ticket, can you troubleshoot the `502 Server Error: Bad Gateway` errors. For example:
- `Error processing CUI C0554107: 502 Server Error: Bad Gateway for url` 
- `Error processing Semantic Group URI for row 1132`

Please update script as needed 
Update readme.md

ZubairL commented:
script has been running for ~ 1 hr  on the Mental Health Diagnosis keyword set, will post updates here once it finished!

ZubairL commented:
Update: mental health keyword set finished and ran. Only got two errors: 
JSONDecodeError encountered for CUI: C0270460. Skipping this entry.
JSONDecodeError encountered for CUI: C0458110. Skipping this entry.

These were the errors that we were seeing before anyway, user should just manually plug these into UMLS.

Going to run on  Eyes- Field Codesets. Will update when finished

ZubairL reviewed (approved):
Eyes keywords works great, approving :)

hannahbarker2 reviewed (approved):
This looks great! Just be sure to resolve any merge conflicts if the other PR is merged first.

amwarnock commented:
Hi @ZubairL, @hannahbarker2  and @jlpeterson2 ! Thank you for your reviews.  Can you please take another look at the script and give me some updated feedback? I've updated the script's error messaging, removed the Children's code call section, and updated the troubleshooting ReadMe page.  I'd like 2 additional pieces of feedback before resolving the merge conflicts and committing this to main.  Thanks!

jlpeterson2 reviewed (approved):
Thanks, @amwarnock This looks great!
